### ISSUE TITLE
ramp() should be refactored

### ISSUE THREAD

**Issue by davidmanzanares** [CartoDB/carto-vl#928]
ramp() is bloated now. It still contains the baggage of the early stages of development, but it also includes tons of functionality. Some can be abstracted, some can be moved to other expressions (without introducing API changes).

We found this while reviewing ramp legends, and we are doing this from that branch.

**Comment by makella** [comment]
Just for clarity, can you provide a few examples of current bloat/baggage that can be abstracted? Curious to know more.

**Comment by davidmanzanares** [comment]
Sure!

Part of it was due to successive `ramp` patches (`ramp` has received lots of changes and improvements) without refactoring old legacy code. Another problem was that it was doing more than it should without relying on the other expressions like `buckets` or `top` for the implementation of features. Lastly, having the `others` color within the array and having many cases to control was a huge pain. With this, we reviewed the original spec and made a minimal simplification (which is a breaking change, but a really minor one), we'll discuss it with more detail in the PR.
